shared/telemetry/README.md

```diff
@@ -0,0 +1,108 @@
+## Welcome to the Telemetry Module
+
+This module is used to collect telemetry data from the node.
+
+At the moment, we are using two types of metrics:
+
+- Time series metrics: these are metrics that are collected periodically and are stored in a database.
+- Event series metrics: these are metrics that are collected when an event occurs and are stored in a database.
+
+
+At the moment, we are using for each of the two types of metrics respectively:
+
+- Prometheus
+- Plain Logs. (_Might be subsituted in the future with an events database_)
+
+
+## Usage
+
+### Time Series Metrics
+
+If you aren't familiar with time series metrics that Prometheus offers, please check out [Prometheus Metrics](https://prometheus.io/docs/concepts/metric_types/)
+
+
+We are primarily using:
+
+- Gauges
+
+We use Gauges to keep track of:
+
+- Blockheight
+- Nodes Online
+
+#### How to use the time series metrics in your code
+
+In your module, make sure you have access to the bus, then use the metrics you need as follows:
+```go
+
+timeseriesTelemetry := module.GetBus().GetTelemetryModule().GetTimeSeriesAgent()
+// explore the methods you can use in shared/modules/telemetry_module.go
+
+// To increment a gauge
+timeseriesTelemetry.IncGauge("gauge_name", 1)
+// etc...
+```
+
+### Event Metrics
+
+In the current implementation, we are recording events through logs. For each unique witnessing of a given event, we will emit a log line with specific formatting to stdout.
+
+Using Loki and Grafana, we parse the logs and generate the desired graphs.
+
+In your module, make sure you have access to the bus, then use the metrics you need as follows:
+```go
+
+timeseriesTelemetry := module.GetBus().GetTelemetryModule().GetTimeSeriesAgent()
+// explore the methods you can use in shared/modules/telemetry_module.go
+
+eventMetricsTelemetry.EmitEvent(
```

Comment:
Why not enforce `namespace`, `event_name` followed by optional args in `EmitEvent`?

Reply:
Affirmative. +1